net_http_bsd/src/main.c

```diff
@@ -49,38 +55,67 @@ unsigned char readUTF(unsigned char c, int fd){
 	case 9516:return 0x12;
 	case 9524:return 0x11;
 	case 9532:return 0x15;
-	case 9600:return 0xDF;
-	case 9604:return 0xDC;
+	case 9600:return 0xC0;
+	case 9601:return 0xC0;
+	case 9602:return 0xDC;
+	case 9603:return 0xDC;
+	case 9604:return 0xC2;
+	case 9605:return 0xC2;
+	case 9606:return 0xDB;
+	case 9607:return 0xDB;
+	case 9608:return 0xDB;
 	}
 	return '?';
 }
 
 int main (int argc, char *argv[]){
-	static char net_mem[1*1024*1024];
+	(void)argc;(void)argv;
+	char* *url=NULL, *wttr[]={"wttr.in","/Paris"}, *rate[]={"rate.sx","/"};
 
-	#ifdef __vita__
 	psvDebugScreenInit();
 	psvDebugScreenFont.size_w-=1;//narrow character printing
+	printf("Press [L]=%s%s [R]=%s%s\n", wttr[0], wttr[1], rate[0], rate[1]);
+	#ifdef __vita__
+	static char net_mem[1*1024*1024];
 	sceSysmoduleLoadModule(SCE_SYSMODULE_NET);
 	sceNetInit(&(SceNetInitParam){net_mem, sizeof(net_mem)});
 	#endif
 
-	char* host = "wttr.in", *path="/Paris";
+	#ifdef __vita__
+	for(SceCtrlData ctrl={}; !url; sceCtrlReadBufferPositive(0,&ctrl,1))
+		switch(ctrl.buttons){
+			case SCE_CTRL_LTRIGGER:url=wttr;break;
+			case SCE_CTRL_RTRIGGER:url=rate;break;
+		}
+	#else
+	while(!url)
+		switch(getchar()){
+			case 'L':case 'l':url=wttr;break;
+			case 'R':case 'r':url=rate;break;
+		}
+	#endif
+
+	printf("fetching %s%s...\n", url[0], url[1]);
 	int pos=0, fd = socket(PF_INET, SOCK_STREAM, 0);
-	connect(fd, (const struct sockaddr *)&((struct sockaddr_in){.sin_family = AF_INET, .sin_port = htons(80), .sin_addr.s_addr = *(long*)(gethostbyname(host)->h_addr)}), sizeof(struct sockaddr_in));
+	connect(fd, (const struct sockaddr *)&((struct sockaddr_in){.sin_family = AF_INET, .sin_port = htons(80), .sin_addr.s_addr = *(long*)(gethostbyname(url[0])->h_addr)}), sizeof(struct sockaddr_in));
 	
-	char*header[] = {"GET ",path," HTTP/1.1\r\n", "User-Agent: curl/7.52.1\r\n", "Host: ",host,"\r\n", "\r\n", 0};
+	char*header[] = {"GET ",url[1]," HTTP/1.1\r\n", "User-Agent: curl/PSVita\r\n", "Host: ",url[0],"\r\n", "\r\n", 0};
 	for(int i=0;header[i];i++)write(fd, header[i], strlen(header[i]));
 
-	for(unsigned char c, line[4096];read(fd,&c,sizeof(c))>0 && pos<sizeof(line);pos = (c=='\n')? 0 : (pos + 1)){
-		#ifdef __vita__
+	for(unsigned char c, line[4096];read(fd,&c,sizeof(c))>0 && pos<(int)sizeof(line);pos = (c=='\n')? 0 : (pos + 1)){
 		if (c>>6==3)c = readUTF(c,fd);
-		#endif
-		if (c == '\n') printf("%.*s\n",pos,line);
+		if (c == '\n') psvDebugScreenPrintf("%.*s\n",pos,line);
 		else line[pos] = c;
 	}
-
 	close(fd);
+
+	#ifndef __vita__ // generate a RGB screen dump if built on PC
+	//convert the dump into PNG with: convert -depth 8 -size 960x544+0 RGB:screen.data out.png;
+	int fdump = open("screen.data", O_WRONLY | O_CREAT | O_TRUNC, 0777);
+	for (int i=0;i<(int)sizeof(base);i += 4)write(fdump, base+i, 3);
```

Comment:
Please expand this loop. It's hard to read, especially for beginner developers who are probably the main target of the samples.